src/components/Input/Input.js

```diff
@@ -1,166 +1,53 @@
-  }
-  .text__error-left & {
-    border-color: ${colors.alert.base};
-  }
-  &.text--input-left {
-    flex-grow: 2;
-  }
-  &:hover {
-    border: 2px ${colors.azure.base} solid;
-    padding-left: 17px;
-  }
-  &:-webkit-input-placeholder {
-    color: ${colors.onyx.light};
-  }
-  &:-moz-placeholder {
-    color: ${colors.onyx.light};
-  }
-  &:focus {
-    border: 1px ${colors.azure.base} solid;
-    padding-left: 18px;
-    border-radius: ${constants.borderRadius.small};
-    box-shadow: 0 0 5px 0 ${colors.azure.base};
-    background-color: ${colors.white.base};
-    outline: none;
-  }
-`;
-const ErrorBoxWrapper = styled.div`
-  display: flex;
-  flex-direction: column;
-  flex-grow: 2;
-`;
-const FieldInputWrapper = styled.div`
-  display: flex;
-  position: relative;
-  &.text--input-top {
-    flex-direction: column;
-    justify-content: left;
-  }
-  &.text--input-left {
-    flex-direction: row;
-  }
-  &.text--input-disabled {
-    pointer-events: none;
-    opacity: 0.35;
-  }
-`;
 
-const FieldInputText = styled.label`
-  font-size: 12px;
-  height: 13px;
-  line-height: 1.25;
-  text-align: left;
-  color: ${colors.onyx.muted};
-  position: relative;
-  .text--input-disabled & {
-    opacity: 0.35;
-  }
-  .text--input-top & {
-    margin-bottom: 6px;
-  }
-  .text--input-left & {
-    margin-right: 11px;
-    margin-top: 10px;
-  }
-  &.text__error-top {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-    font-size: 12px;
-    line-height: 1.25;
-  }
-  &.text__error-left {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-  }
-`;
-
-const FieldErrorText = styled.label`
-  font-size: 12px;
-  height: 13px;
-  line-height: 1.25;
-  text-align: left;
-  color: ${colors.onyx.muted};
-  position: relative;
-  .text--input-disabled & {
-    opacity: 0.35;
-  }
-  .text--input-left & {
-    margin-right: 11px;
-    margin-top: 10px;
-  }
-  &.text__error-top {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-    font-size: 12px;
-    line-height: 1.25;
-  }
-  &.text__error-left {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-  }
-`;
+import sluggify from "../../utils/text";
+import INPUT_SIZES from "./constants";
+import {
+  FieldInputWrapper,
+  FieldInputBox,
+  ErrorBoxWrapper,
+  FieldInputText,
+  FieldErrorText
+} from "./Input.styles";
 
 const Input = ({
   labelPosition,
   placeholder,
   disabled,
-  width,
   label,
   name,
   errorMessage,
+  size,
   ...rest
 }) => (
   <FieldInputWrapper
     className={classNames({
       "text--input-top": labelPosition === "top",
       "text--input-left": labelPosition === "left",
       "text--input-disabled": disabled,
-      "text__error-top": labelPosition === "top" && errorMessage,
-      "text__error-left": labelPosition === "left" && errorMessage
+      text__error: errorMessage,
+      "text--input-small": size === "small",
+      "text--input-regular": size === "regular",
+      "text--input-large": size === "large"
```

Comment:
You could replace the labelPosition-related classnames with `[`text--input-${labelPosition}`]: !!labelPosition`
You could replace the size-related classnames with `[`text--input-${size}`]: !!size`

Reply:
I think I can do
```
`text--input-${size}`,
`text--input-${labelPosition}`,
{
  "text--input-disabled": disabled,
  text__error: errorMessage,
}
```
Thanks for the hint.